app/controllers/books_controller.rb

```diff
@@ -96,6 +103,38 @@ def show_more
     render json: books, status: 200
   end
 
+  def fill_language_id(book)
+    # when the source_language_id is not set, try to auto-fill it
+    if book.source_language_id.length == 0
+      res_id = id_check(book.source_language)
+      if res_id != nil
+        book.source_language_id = res_id
+      end
+    end
+
+    # when the target_language_id is not set, try to auto-fill it
+    if book.target_language_id.length == 0
+      res_id = id_check(book.target_language)
+      if res_id != nil
+        book.target_language_id = res_id
+      end
+    end
+
+    return book
+  end
+
+  # given a identifier string, get the language ID
+  # only returns an ID if the verbatim search returns exactly one result
+  def id_check(input_lang)
+    # TODO do not hard-code this URL, move to a config constant
+    url_string = 'http://localhost:6543/search?q=' + input_lang + '&whole=true';
```

Comment:
I wasn't sure how to properly set up a config constant. Ideally this constant can be used by both the Ruby code and the React code.

Reply:
We can make an Env variable through Heroku. Happy to do together if you want